src/test/netbase_tests.cpp

```diff
@@ -158,6 +158,19 @@ BOOST_AUTO_TEST_CASE(embedded_test)
     BOOST_CHECK_EQUAL(addr1.ToStringAddr(), addr2.ToStringAddr());
 }
 
+BOOST_AUTO_TEST_CASE(dns_lookup)
+{
+    CNetAddr addr;
+    bool ret = LookupHost("nic.com", addr, /*fAllowLookup=*/true);
+    BOOST_CHECK(ret);
+    BOOST_CHECK(addr.IsRoutable());
```

Comment:
Making actual requests from the unit tests is not great.

If you want to add a unit test, you could introduce a way to mock `getaddrinfo` or `AsyncGetAddrInfo` so that you can better test our code in all the relevant scenarios (i.e. getaddrinfo errors/blocking etc.).